### DISCUSSION TITLE
Latex support

### DISCUSSION THREAD

**Discussion by h7b** [ObsidianPublisher/obsidian-github-publisher#165] [Q&A — Template] [answered]
Hi Mara,

## What's wrong

I saw in plugin docs page that the template already supports Latex [^1]. However the tex code did not render properly in my case (screencap below). If you have any idea about what's possibly wrong, please give me some insights so I can investigate further.

![Screenshot 2023-06-13 235148](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/86811584/67484cc4-24fa-4a27-b00e-a89c0c7b5a7c)

## What did I try

As my understanding from [mkdocs-material docs](https://squidfunk.github.io/mkdocs-material/reference/math/#mkdocsyml), you enabled the MathJax by default for latex rendering in your [template repo](https://github.com/ObsidianPublisher/template-netlify-vercel).

I also tried to enable Katex following the [mkdocs-material docs](https://squidfunk.github.io/mkdocs-material/reference/math/#katex). But the tex code is still not being rendered correctly

For your reference if you are curious enough, please find below
- link to wrongly-rendered latex published note: [note.html](https://h7b1.netlify.app/unsorted/scenario-declared-salary-payslip/)
- link to the `mkdocs.yml` settings in my note repo: [mkdocs.yml](https://github.com/h7b/h7b-public-mkdocs/blob/main/mkdocs.yml)
- link to specific line with latex code in the markdown note upload to repo: [note.md](https://github.com/h7b/h7b-public-mkdocs/blob/3dfd1c5f6d77fc751db552afb3f2dacec8b92bc0/docs/unsorted/scenario-declared-salary-payslip.md?plain=1#LL43C1-L44C1)

[^1]: https://obsidian-publisher.netlify.app/plugin/?h=latex#features

**Comment by Mara-Li** [comment]
I dunno, to be honest. I think you need to contact / make a bug report or discussion on the Official Material Mkdocs repository. 

On my side, Latex works properly so... 
I think you will gain some information on this repo more than mine!

**Reply by h7b** [reply]
Thank Mara for your info. 

I was in hope that it might be a regression bug when u modified the template. Since I noticed that all the `$`...`$` delimiter for latex block code were replaced by `\(`...`\)`, which is weird.

But it works on your side, so it is not in this case. I will try to spin up a new repo to see if the latex works. I also noticed that you have changed the way to update template with the new "actions" repo. Still reading your written docs on how to implement the workflow.

**Reply by Mara-Li** [reply, marked as answer]
Note that I changed the Javascript loading order, so in `extra_javascript` you need to change the order and set ` https://cdn.jsdelivr.net/gh/ObsidianPublisher/assets@main/dist/javascript.js` in first.

Like that: 
```yaml
  - https://cdn.jsdelivr.net/gh/ObsidianPublisher/assets@main/dist/javascript.js
  - https://polyfill.io/v3/polyfill.min.js?features=es6
  - https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js
  - assets/js/snippets.js
 ```

**Reply by h7b** [reply]
Thank you Mara.

Exactly this reason.

## What solved my issue?

In `extra_javascript` I change the order of `https://cdn.jsdelivr.net/gh/ObsidianPublisher/assets@main/dist/javascript.js` as you said.

Now I can remove the custom `katex.js` from my repo.
